app/functions_man.py

```diff
@@ -0,0 +1,100 @@
+import time
+from bs4 import BeautifulSoup
+from selenium.webdriver.common.by import By
+from categories_man import driver
+
+
+def find_index(tag, class_, clothesBS):
+    index_text = clothesBS.find(tag, class_).text
+    index_text_list = index_text.split(" ")
+    index = index_text_list[-1]
+    return index
+
+
+def find_price(clothesBS):
+    span = clothesBS.find("div", "product-detail-info__price-amount price")
+    price_with_currency = span.find("span", "price-current__amount").text
+    price_text = price_with_currency.split(" ")[0]
+    if len(price_text) > 6:
+        price_text = price_text.replace("\xa0", "")
+    price = float(price_text.replace(",", "."))
+    return price
+
+
+def find_color(clothesBS):
+    if (
+        clothesBS.find(
+            "div", "product-detail-color-selector product-detail-info__color-selector"
+        )
+        != None
+    ):
+        color_div = clothesBS.find(
+            "div",
+            "product-detail-color-selector product-detail-info__color-selector",
+            clothesBS,
+        )
+        color_span = color_div.find_all("span", "screen-reader-text")
+        color_list = []
+        for span in color_span:
+            color = span.text
+            color_list.append(color)
+        return color_list
+    else:
+        color_text = clothesBS.find(
+            "p", "product-detail-selected-color product-detail-info__color"
+        ).text
+        color_list = color_text.split(" ")
+        color = color_list[0]
+        try:
+            color_int = int(color[0])
+            color = []
+        except ValueError:
+            color = [color]
+        return color
+
+
+def find_prices_colors(category, clothesBS, path):
+    for span in clothesBS.find_all("div", "product-detail-info__price-amount price"):
+        old_price_span = span.find(
+            "span", "price-old__amount price__amount price__amount-old"
+        )
+        if old_price_span != None:
+            old_price_with_currency = old_price_span.text
+            old_price_text = old_price_with_currency.split(" ")[0]
+            old_price = float(old_price_text.replace(",", "."))
+            price = find_price(clothesBS)
+            colors = find_color(clothesBS)
+            return [old_price, price, colors]
+
+        elif old_price_span == None and category != "promocje":
+            old_price = None
+            colors = find_color(clothesBS)
+            price = find_price(clothesBS)
+            return [old_price, price, colors]
+        else:
+            driver.get(path)
+
+            time.sleep(6)
```

Comment:
avoid hardcode values, try to change it to variable

Reply:
I added waiting_time